include/misc/slist.h

```diff
@@ -38,6 +39,7 @@ struct _slist {
 
 typedef struct _slist sys_slist_t;
 
+
```

Comment:
Nitpick: unrelated change and redundant empty line

I'm curious to see how the new list variants will look like in practice!